node-authorizer/pkg/server/middleware.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"net/http"
+
+	"k8s.io/kops/node-authorizer/pkg/utils"
+
+	"go.uber.org/zap"
+)
+
+// authorized is responsible for validating the client certificate
+func authorized(next http.HandlerFunc, commonName string, requireAuth bool) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if requireAuth {
```

Comment:
I'm wondering if it would be safer just to have one function with all the steps, just so the processing is explicit, rather than using filters & the chain.  In practice you'd just call out to e.g. `if checkClientCertificates && !hasValidClientCertificate(...)` etc - I'm not suggesting inlining all the code :-)

Reply:
So the thinking behind this was to never pass a `*http.Request` into the business logic _(or admission logic)_ it's passed `NodeRegistration{}` only which felt cleaner.